### PR TITLE
CORE-6958: Ensure we find all non-injectable sandbox services.

### PR DESCRIPTION
**Pull request by chrisr3**
Fix bug affecting how sandboxes find non-injectable services.

### CODE REVIEW

File: components/virtual-node/sandbox-group-context-service/src/main/kotlin/net/corda/sandboxgroupcontext/service/impl/SandboxGroupContextServiceImpl.kt [hunk 1]
```diff
@@ -182,9 +183,19 @@ class SandboxGroupContextServiceImpl(
...
-                            singleton(serviceRef.bundle).loadCommonService(serviceType)
+                            // We always load interfaces for system services.
+                            serviceRef.loadCommonService(serviceType)
                         } else if (accessControlContext.checkServicePermission(serviceType)) {
-                            bundles.loadCommonService(serviceType)
+                            // Non-system services must be allowed by the security policy.
+                            if (vnc.sandboxGroupType == SandboxGroupType.PERSISTENCE) {
+                                // PERSISTENCE sandboxes do not execute code, so accept
+                                // every service that is allowed by the security policy.
+                                serviceRef.loadCommonService(serviceType)
```
Comment (driessamyn) [inline components/virtual-node/sandbox-group-context-service/src/main/kotlin/net/corda/sandboxgroupcontext/service/impl/SandboxGroupContextServiceImpl.kt]: Is this still true?
Speaking to @owenstanford yesterday, it sounds like the "services replacement" need to be able to run hooks in a persistence sandbox.

Comment (chrisr3) [reply components/virtual-node/sandbox-group-context-service/src/main/kotlin/net/corda/sandboxgroupcontext/service/impl/SandboxGroupContextServiceImpl.kt]: Are these hooks _our_ code, though? But anyway, what I've actually done here is _remove_ the requirement that the service have a bundle wiring to the sandbox, which effectlvely means `PERSISTENCE` sandboxes receive every `UsedByPersistence` service with a `net.corda.v5.*` interface.

Comment (owenstanford) [reply components/virtual-node/sandbox-group-context-service/src/main/kotlin/net/corda/sandboxgroupcontext/service/impl/SandboxGroupContextServiceImpl.kt]: No the hooks will be user code. they will require injection similar to flows but with a limited set of services (initially just the marshalling services).

Comment (chrisr3) [reply components/virtual-node/sandbox-group-context-service/src/main/kotlin/net/corda/sandboxgroupcontext/service/impl/SandboxGroupContextServiceImpl.kt]: Out of interest, which injectable services other than `JsonMarshallingService` are we talking about here?

**Review by driessamyn — APPROVED** [review]
LGTM

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**